base/resources/scripts/languagecheck.js

```diff
@@ -0,0 +1,40 @@
+/* Nodejs to set up a server on 8081 to modify the argv.json file
+   used by in-browser VSCode. Uses the accept-language header.
+   Also used by jupyter to determine language.
+ */
+var http = require("http");
+var fs = require("fs");
+
+http
+  .createServer(function (req, res) {
+    res.writeHead(200, { "Content-Type": "text/html" });
+    var lang = req.headers["accept-language"];
+    lang ? lang : (lang = "en");
+    lang = lang.substring(0, 2);
+
+    lang === "fr" || lang === "en" ? lang : (lang = "en");
```

Comment:
Clever, and succinct, but the assignment expressions and unused expressions make this difficult to understand. Let's go with something like:
```js
// Some constants at the top
const SUPPORTED_LANGS = [...];
const DEFAULT_LANG = "en";

// Then down here
let lang = (req.headers[...] || DEFAULT_LANG).substring(0, 2);
if (!SUPPORTED_LANGS.includes(lang)) {
  lang = DEFAULT_LANG;
}
```